analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicCheckList.cs

```diff
@@ -60,19 +60,31 @@ public void ExecutionCompleted()
         }
 
         public SymbolicContext[] PreProcess(SymbolicContext context) =>
-            InvokeChecks(context, (check, context) => check.PreProcess(context));
+            InvokeChecks(context, preProcess: true);
 
         public SymbolicContext[] PostProcess(SymbolicContext context) =>
-            InvokeChecks(context, (check, context) => check.PostProcess(context));
+            InvokeChecks(context, preProcess: false);
 
-        private SymbolicContext[] InvokeChecks(SymbolicContext context, Func<SymbolicCheck, SymbolicContext, ProgramState[]> process)
+        private SymbolicContext[] InvokeChecks(SymbolicContext context, bool preProcess)
         {
-            var contexts = new[] { context };
+            var before = new List<SymbolicContext>() { context };
```

Comment:
This will need a performance related comment so we're not tempted to refactor it too much